docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
-        });
-        const resp = await DataSetDiff.diff(session, params.arguments.oldDataSet, params.arguments.newDataSet);
-        params.response.console.log(resp);
-    }
-}
+3. In the `src` directory, add the following code to the end of the `index.ts` file to make the API available for other developers to import:
 
-```
+    ```typescript
+    export * from "./api/DataSetDiff";
+    ```
+
+    When successful, **[WHAT IS THE USER SUPPOSED TO SEE?]**.
+
+**To create a unique plug-in:** The file name and code in Step 2 may be entirely different if you want to implement an API to do something else.
+
+## Building your plug-in source
+
+Confirm that your project builds successfully.
+
+Follow these steps:
+
+1. In the terminal, run `npm run build` to verify there are no errors.
+
+    This command builds your typescript files into javascript files for your plug-in in the `lib` directory.
+
+2. If you come across linting errors, run `npm run lint -- --fix` to resolve the errors automatically.
+
+    When successful, no errors are returned and the `lib` directory contains the built javascript files.
+
+**To create a unique plug-in:** Follow these same steps.
+
+## Creating a Zowe CLI command
+Define the command that calls the client-side API.
+
+Follow these steps:
+
+1. In `src/cli`, create a folder named `diff`.
+2. In the `diff` directory, create a file named `Diff.definition.ts`.
+3. Copy and paste the following code into the `Diff.definition.ts` file:
+
+    ```typescript
+    import { ICommandDefinition } from "@zowe/imperative";
+    import { DataSetsDefinition } from "./data-sets/DataSets.definition";
+    const IssueDefinition: ICommandDefinition = {
+        name: "diff",
+        summary: "Diff two data sets content",
+        description: "Uses open source diff packages to diff two data sets content",
+        type: "group",
+        children: [DataSetsDefinition]
+    };
+
+    export = IssueDefinition;
+    ```
+
+4. In the `diff` folder, create a folder named `data-sets`. 
+5. In the `data-sets` folder, create the following two files: 
+    - `DataSets.definition.ts`
+    - `DataSets.handler.ts`
+
+6. Copy and paste the following code into the `DataSets.definition.ts` file:
+    ```typescript
+    import { ICommandDefinition } from "@zowe/imperative";
+
+    export const DataSetsDefinition: ICommandDefinition = {
+        name: "data-sets",
+        aliases: ["ds"],
+        summary: "data sets to diff",
+        description: "diff the first data set with the second",
+        type: "command",
+        handler: __dirname + "/DataSets.handler",
+        positionals: [
+            {
+                name: "oldDataSet",
+                description: "The old data set",
+                type: "string"
+            },
+            {
+                name: "newDataSet",
+                description: "The new data set",
+                type: "string"
+            }
+        ],
+        profile: {
+            required: ["zosmf"]
```

Comment:
should this profile be `optional: ["zosmf"]` to follow Zowe V2 conformance and best practices?